qemu/tests/block_hotplug.py

```diff
@@ -88,153 +75,83 @@ def unplug_device(vm, get_disk_cmd, device):
 
     for iteration in range(repeat_times):
         device_list = []
-        controller_list = []
-        controller_device_dict = {}
         error_context.context("Hotplug block device (iteration %d)" % iteration,
                               logging.info)
 
-        sub_type = params.get("sub_type_before_plug")
-        if sub_type:
-            error_context.context(context_msg % (sub_type, "before hotplug"),
-                                  logging.info)
-            utils_test.run_virt_sub_test(test, params, env, sub_type)
+        plug_tag = "before_plug"
+        run_sub_test(params, plug_tag)
 
         for num in range(blk_num):
-            device = qdevices.QDevice(pci_type)
+            image_name = img_list[num + 1]
+            image_params = params.object_params(image_name)
+            drive_format = image_params.get("drive_format")
             if params.get("need_plug") == "yes":
                 disks_before_plug = find_disk(vm, get_disk_cmd)
+                devs = vm.devices.images_define_by_params(image_name,
+                                                          image_params, 'disk')
+                for dev in devs:
+                    vm.devices.simple_hotplug(dev, vm.monitor)
```

Comment:
Well I'd be more comfortable if:

```python
                    ret = vm.devices.simple_hotplug(dev, vm.monitor)
                    if ret[1] is not True:
                        raise RuntimeError("Unable to verify status of "
                                           "hotplugging '%s'. Ouptut:\n%s"
                                           % (dev, ret[0]))
```

was used to not to mask broken hotplugs (yes, it should fail later, so I leave it on your reconsideration. Either way is fine by me)

Reply:
No QRHDrive.verify_hotplug() defined, when we execute device.verify_hotplug() for QRHDrive device, QBaseDevice.verify_hotplug() will be executed actually, which will return {} finally.
     class QBaseDevice():
         def verify_hotplug(self, out, monitor):
             return out 

For other devices, verify_hotplug() will return True if success, else False.

So based on present qdevices.py design, we suggest to adopt:
     if ret[1] is False:
         raise RuntimeError("Unable to verify status of hotplugging '%s'. "
                                        "Ouptut:\n%s " % (dev, ret[0]))

Reply:
Yep, makes sense. Perhaps something like:

```
if ret[1] is False:
    raise...
elif ret[1] is not True:
   logging.warning("Unable to verify hotplug status: %s...")
```

but perhaps it's unnecessary verbose... I mean when it fails we won't get the disks detected and we'll check the output...